package/src/components/ChannelList/hooks/listeners/useNewMessage.ts

```diff
@@ -34,27 +35,22 @@ export const useNewMessage = <
       } else {
         setChannels((channels) => {
           if (!channels) return channels;
+          const channelInList = channels.filter((channel) => channel.cid === event.cid).length > 0;
 
-          if (!lockChannelOrder && event.cid && event.channel_type && event.channel_id) {
-            const targetChannelIndex = channels.findIndex((c) => c.cid === event.cid);
-
-            if (targetChannelIndex >= 0) {
-              return moveChannelUp<StreamChatGenerics>({
-                channels,
-                cid: event.cid,
-              });
-            }
-
+          if (!channelInList && event.channel_type) {
             // If channel doesn't exist in existing list, check in activeChannels as well.
             // It may happen that channel was hidden using channel.hide(). In that case
             // We remove it from `channels` state, but its still being watched and exists in client.activeChannels.
             const channel = client.channel(event.channel_type, event.channel_id);
-
-            if (channel.initialized) {
-              return [channel, ...channels];
-            }
+            return uniqBy([channel, ...channels], 'cid');
```

Comment:
uniq is unnecessaruy isnt it? we are adding a new channel here..

Reply:
Fixed